Improve logs around network communication

luckysori opened the pull request:
We have several instances were we are:

- Logging the same thing more than once (objective).
- Logging at the wrong level (subjective).

I've stated several times that I believe we should be more selective with our usage of the `ERROR` level. In my mind, if I see an error log I should immediately understand that there's something to investigate. If I see a single warning I should be able to safely ignore it. And if I see many warnings of the same kind, I should start to suspect that there's a real problem.

da-kami commented on model/src/cfd.rs:
```diff
@@ -1328,7 +1328,7 @@ impl Cfd {
 
     pub fn fail_collaborative_settlement(self, error: anyhow::Error) -> CfdEvent {
-        tracing::error!(order_id=%self.id(), "Collaborative settlement failed: {:#}", error);
+        tracing::warn!(order_id=%self.id(), peer_id=?self.counterparty_peer_id, "Collaborative settlement failed: {:#}", error);
```
🙃 I think so far we mostly opted for keeping the log in the `protocol` rather than in `model::Cfd`. I am not sure what is better, but we should probably agree on one or the other.

For me the logs should be part of the protocol actors, because a model should not log anything for me.

luckysori replied:
But this PR doesn't add any logs to `model::cfd`. It's only modifying one instance out of 19 pre-existing log statements that were already there.

I'm not against keeping the `model` crate effect-free, but that's not something that we've actually been doing since we introduced it.

da-kami replied:
yeah, totally. I was just wondering if we should rather remove the logs from model crate and keep them in the other crates. It's probably something that we can discuss and then add a specific ticket. Feel free to ignore it for this PR :)

Restioson replied:
Nit on this line: do we want to have some kind of consistent style with logs and tags? Are we going with no spaces between label and value or more like `order_id = %self.id()`

luckysori replied:
> Nit on this line: do we want to have some kind of consistent style with logs and tags? Are we going with no spaces between label and value or more like order_id = %self.id()

Good catch. I like having spaces too, I'll fix it. But anything that cannot be automatically enforced by `rustfmt` is just too hard to keep under control :(

da-kami commented on xtra-libp2p-rollover/src/v_2_0_0/maker.rs:
```diff
@@ -135,11 +135,10 @@ where
         {
             Ok(base_dlc_params) => base_dlc_params,
             Err(e) => {
-                tracing::warn!(%order_id, "Rollover failed after handling taker proposal: {e:#}");
```
Same: Not sure if it would not be better to let the protocol be in charge of the log rather than the model.

da-kami reviewed (approved):
Thank you for tackling this and removing all the duplicates.
I agree that all the warning logs should be warning logs, but I think our warning load will go up significantly with them. Noting to action here yet.

Feel free to ignore my comment. I am not sure if we should default to logging in the model rather than the protocol. My intuition says the protocol is actually more appropriate, but I don't have especially good arguments for that.

Restioson commented on daemon/src/collab_settlement/maker.rs:
```diff
@@ -156,7 +155,9 @@ impl Actor {
                     let DialerSignature { dialer_signature } = framed
                         .next()
-                        .timeout(SETTLEMENT_MSG_TIMEOUT, || tracing::debug_span!("receive dialer signature"))
+                        .timeout(SETTLEMENT_MSG_TIMEOUT, || {
```
Are these changes from dprint or manual?

luckysori replied:
It was `dprint`. 

Funnily enough it only happens when I save the files using emacs, which calls `dprint` via [`dprint-fmt`](https://github.com/luckysori/dprint-fmt).

Sorry about the diff noise!

Restioson replied:
Huh, that's a bit weird :thinking: I wonder how it originally passed CI without being properly formatted...

luckysori replied:
My guess is that the workspace-level call to `dprint check` is okay with the file either way.

And the [file-level call](https://github.com/luckysori/dprint-fmt/blob/88ab37467a4a7a7d7660297b462b676239508485/dprint-fmt.el#L50) to `dprint fmt -stdin <file-path>` using the `dprint-fmt` emacs package sees things differently here, which is very weird but mostly harmless.

da-kami commented on xtra-libp2p-offer/src/maker.rs:
```diff
@@ -45,7 +45,7 @@ impl Actor {
 
         let err_handler =
-            move |e| async move { tracing::debug!(%peer, "Failed to send offers: {e:#}") };
+            move |e| async move { tracing::warn!(%peer, "Failed to send offers: {e:#}") };
```
We should totally have these on `warn`. I just wanted to compare the behavior of my `mainnet` taker with the maker logs and just don't have these logs because they are currently on debug and the default log level is info 😅

bonomat reviewed (approved):
Thanks for the clean-ups.

luckysori committed:
Remove duplicate log message in collab settlement

The same message will eventually be emitted as a warning inside
`model/cfd`.

To not lose the `peer` context we add it to the log message inside
`model/cfd`. It is an optional field, but if it's called by the
`/itchysats/collab-settlement` protocol, it will be present.

luckysori committed:
Warn if collab settlement rejection cannot be delivered

It is not a crucial thing to miss, but if we are consistently unable
to tell the taker that we want to reject collaborative settlement, it
could be a sign that there's an underlying problem.

luckysori committed:
Use full error when collaborative settlement fails

Calling `emit_failed` leads to logging a warning in `model/cfd`. We
can save ourselves some repetitive log lines if we pass the full error
context to `emit_failed`.

luckysori committed:
Warn if offers cannot be delivered

It is not a crucial thing to miss once, but if we are consistently
unable to broadcast offers, it should be a sign that there's an
underlying problem.

luckysori committed:
Warn if offers cannot be received

It is not a crucial thing to miss once, but if we are consistently
unable to receive offers, it should be a sign that there's an
underlying problem.

luckysori committed:
Warn if ping cannot be received

It is not a crucial thing to miss once, but if we are consistently
unable to receive pings, it should be a sign that there's an
underlying problem.

luckysori committed:
Warn if rollover rejection cannot be delivered

It is not a crucial thing to miss, but if we are consistently unable
to tell the taker that we do not want to rollover, it could be a sign
that there's an underlying problem.

luckysori committed:
Remove duplicate log message in maker rollover

The same message will eventually be emitted as a warning inside
`model/cfd`.

By adding the message using `anyhow::Context`, we ensure that the
inner warning talks about the failure to handle the taker's proposal.

luckysori committed:
Simplify warning about failure to send revocation keys

The extra information belongs in a comment. Otherwise the log message
gets needlessly verbose.

luckysori committed:
Remove log duplicates in rollover protocol

All these messages are already emitted in `model`.

luckysori committed:
Remove duplicate log message in taker rollover

The same message will eventually be emitted as a warning inside
`model/cfd`.

By adding the message using `anyhow::Context`, we ensure that the
inner warning talks about the "start" of the rollover.
